packages/floating-label/index.tsx

```diff
@@ -0,0 +1,140 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import * as React from 'react';
+import * as classnames from 'classnames';
+// @ts-ignore
+import {MDCFloatingLabelFoundation} from '@material/floating-label';
+
+
+export interface FloatingLabelProps {
+  className?: string,
+  handleWidthChange?: (width: number) => void,
+  float?: boolean,
+};
+
+type Props<T> = FloatingLabelProps & React.LabelHTMLAttributes<T>;
+
+type FloatingLabelState = {
+  classList: Set<String>,
+};
+
+export default class FloatingLabel extends React.Component<
+  Props<HTMLLabelElement>,
+  FloatingLabelState
+  > {
+  foundation_?: MDCFloatingLabelFoundation;
+  labelElement: React.RefObject<HTMLLabelElement> = React.createRef();
+
+  static defaultProps: Partial<Props<HTMLLabelElement>> = {
```

Comment:
we don't need a default for handleWidthChange, since it is just undefined.


I'm going to change to Props<HTMLLabelElement> because that is the props going into the component. I'm renaming to maybe make more sense. LMK what you think.